### PR TITLE
require the kernel image to still exist before allowing hibernation

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by poettering** [comment]
The first commit is a fix-up for db69869f264af2d1afcdd3e573e0e9fdd5bef065, which @falconindy originally added, and which looks a lot like a type. I haven't tested this however, hence could you, @falconindy comment?

**Comment by falconindy** [comment]
Seems

File: src/login/logind-dbus.c [hunk 1]
```diff
@@ -1770,11 +1770,11 @@ static int method_do_shutdown_or_sleep(
         if (sleep_verb) {
                 r = can_sleep(sleep_verb);
                 if (r == -ENOSPC)
-                        return sd_bus_error_set(error, BUS_ERROR_SLEEP_VERB_NOT_SUPPORTED,
-                                                "Not enough swap space for hibernation");
+                        return sd_bus_error_set(error, BUS_ERROR_SLEEP_VERB_NOT_SUPPORTED, "Not enough swap space for hibernation");
+                if (r == -ENOMEDIUM)
+                        return sd_bus_error_set(error, BUS_ERROR_SLEEP_VERB_NOT_SUPPORTED, "Kernel image has been removed, can't hibernate");
```
Comment (keszybz) [inline src/login/logind-dbus.c:1775]: "The CD-ROM with the kernel has been ejected" :)

File: src/shared/sleep-config.c [hunk 2]
```diff
@@ -257,6 +260,72 @@ static bool enough_swap_for_hibernation(void) {
...
 
+static int kernel_exists(void) {
+        struct utsname u;
+        sd_id128_t m;
+        int i, r;
+
+        /* Doe some superficial checks whether the kernel we are currently running is still around. If it isn't we
+         * shouldn't offer hibernation as we couldn't possible resume from hibernation again. Of course, this check is
+         * very superficial, as the kernel's mere existance is hardly enough to know whether the hibernate/resume cycle
+         * will succeed. However, the common case of kernel updates can be caught these way, and it's definitely worth
```
Comment (keszybz) [inline src/shared/sleep-config.c]: Typo: this

I would be nice to hook this up in `test-sleep`, but that can be done later.

**Comment by keszybz** [comment]
I just merged the other PR that touches the same path (and where "partition" is actually correct), so this one needs adjusting...

**Comment by poettering** [comment]
force pushed an updated version, addressing your points. nothing else. PTAL.

**Pull request merged**